heron/api/src/java/org/apache/heron/streamlet/impl/BuilderImpl.java

```diff
@@ -61,6 +62,14 @@ public BuilderImpl() {
     return retval;
   }
 
+  @Override
+  public <R> Streamlet<R> newSource(IRichSpout spout) {
+    StreamletImpl<R> retval = StreamletImpl.createSpoutStreamlet(spout);
+    retval.setNumPartitions(1);
```

Comment:
Number of partitions is already set at createSpoutStreamlet#SpoutStreamlet level so this looks redundant. 
Also, same for following source functions:
```
newSource(SerializableSupplier<R> supplier)
newSource(Source<R> generator)
```

Reply:
True.